dbtmetabase/logger/logging.py

```diff
@@ -0,0 +1,68 @@
+import sys
+import logging
+from logging.handlers import RotatingFileHandler
+from functools import lru_cache
+from pathlib import Path
+
+from rich.logging import RichHandler
+
+
+# Config Log Files
+LOG_FILE_FORMAT = logging.Formatter(
+    "%(asctime)s — %(name)s — %(levelname)s — %(message)s"
+)
+LOG_PATH = Path.home().absolute() / ".dbt-metabase" / "logs"
+
+# Config Console Log Level
+LOGGING_LEVEL = logging.INFO
+
+
+def rotating_log_handler(
+    logger_name: str,
+) -> RotatingFileHandler:
+    """This handler writes warning and higher level outputs to logs in a home .dbt-metabase directory rotating them as needed"""
+    LOG_PATH.mkdir(parents=True, exist_ok=True)
+    handler = RotatingFileHandler(
+        str(LOG_PATH / f"{logger_name}.log"), maxBytes=int(1e6), backupCount=3
+    )
+    handler.setFormatter(LOG_FILE_FORMAT)
+    handler.setLevel(logging.WARNING)
+    return handler
+
+
+def console_log_handler(
+    level: str,
+) -> logging.StreamHandler:
+    """This handler routes logging output to the console but we are using rich instead"""
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setFormatter(LOG_FILE_FORMAT)
+    handler.setLevel(getattr(logging, level.upper()))
+    return handler
+
+
+@lru_cache(maxsize=128)
```

Comment:
Is the point of caching this, instead of using a static instance, just so the logger name can be applied? Level seems to be statically defined (intentionally or accidentally, considering that it's listed in arguments), so if it's just the name, the function with a cached return seems like an overkill.

You're also not setting any arguments in this project, so it's never going to change from default values.

Reply:
Level is defined so we dont have to pass the value every time but the logger is able to be used with a passed level or different passed name in a specific call without creating multiple logger objects in your code.
Your right that we aren't logging with enough complexity to warrant, so we can definitely simplify this.

Reply:
I parameterized it further and made the code in the logger more understandable. I went to remove the cache but that involved instantiating logger pointers in every module which started to deviate from the original intent; which was to unify and centralize logging to a single functional interface-- modifying logger module attributes anytime during execution for global overrides or providing args for locally scoped overrides on a per call basis. Neither of which can be done with a pointer without mutating the logger handlers locally. 

I know we don't use it for anything too complex today, but complexity isn't the issue -- using the cache here (I downsized it) and using the logger from the single interface `logger().info...` is technically _less_ code than it would be instantiating pointers in each module with logging.getLogger while being more powerful/flexible interface for the future of whoever contributes.